File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema12Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class Schema12Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema08Test::class.java.getResourceAsStream("/schema12.dbs"))
    }

    @Test
    fun testBytearray() {
        val envelope = mapOf("type" to "query", "data" to "abcdefghijklmnopqrstuvwxyz".toByteArray())
        assertMap(engine, envelope, envelope)
    }

}

File: dynabuffers-python/tests/usecase/Schema12Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema12Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))
    error = "int too large to convert"

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(f"{self.root_dir}/schema12.dbs"))

    def test_bytearray(self):
        obj = {"type": 'query', "data": bytearray(b"abcdefghijklmnopqrstuvwxyz")}
        result = self.engine.deserialize(self.engine.serialize(obj))
        self.assertEqual(result, obj)


if __name__ == "__main__":
    unittest.main()
